### PR TITLE
Add a blog post about memory performance improvements

### LINKED ISSUE #51: hook-up blog module
https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/

might make sense to do some posts on here occasionally, when bandwidth allows.

### PR DESCRIPTION
**Pull request by clux**
something i had stashed in my cache for the past couple of weeks.

for https://github.com/kube-rs/kube/issues/1513.
if it looks good i will release 0.92 and edit the post with updated links to 0.92 parts on docs.rs

closes #51 with a first post.

general look (unchanged from upstream module):

grim-area-2024-06-11-23_44_22

### CODE REVIEW

**Review by mateiidavid — COMMENTED** [review]
This looks great! Honestly, nothing blocking from my side. I did leave some nitpicky comments (taking my wordsmithing job seriously) and a suggestion or two.

Thanks for the awesome write-up, and ofc great work to people involved in making this happen.

File: docs/blog/posts/2024-06-11-reflector-memory.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+!!! note "watcher atomicity < 0.92.0"
+
+    You only see a `Restarted` on re-lists once every object has been received through an `api.list`.
+    Watcher events will pause between a de-sync / restart and a `Restarted`. See [watcher::Event@0.91](https://docs.rs/kube/0.91.0/kube/runtime/watcher/enum.Event.html).
+
+This property meant that stores could in-turn provide their own guarantee very easily:
+
+!!! note "Store completeness"
+
+    [Store] always presents the full state once initialised. It present previous full state when initialised and relisting.
```
[the remark this answers is not in the document]
Comment (clux) [reply docs/blog/posts/2024-06-11-reflector-memory.md]: have added a bit h2 about the watch api and justifications now. feels useful to justify this.
https://github.com/kube-rs/website/pull/65/commits/5b717a191e87dd611834a515369e8fa95a77ed4b

File: docs/blog/posts/2024-06-11-reflector-memory.md [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+Improving what is stored in the `Cache` above is important, but it is not the full picture...
+
+## Guarantees
+
+The [watcher] up until 0.92.0 has maintained a guarantee we have casually referred to as __watcher atomicity__:
+
+!!! note "watcher atomicity < 0.92.0"
+
+    You only see a `Restarted` on re-lists once every object has been received through an `api.list`.
+    Watcher events will pause between a de-sync / restart and a `Restarted`. See [watcher::Event@0.91](https://docs.rs/kube/0.91.0/kube/runtime/watcher/enum.Event.html).
```
[the remark this answers is not in the document]
Comment (clux) [reply docs/blog/posts/2024-06-11-reflector-memory.md:48]: have added a bit h2 about the watch api and justifications now. feels useful to justify this.
https://github.com/kube-rs/website/pull/65/commits/5b717a191e87dd611834a515369e8fa95a77ed4b

**Review by mateiidavid — APPROVED** [review]
🚢

**Pull request merged**